src/Pages/Shared/ConfigMapSecret/ConfigMapSecretProtected.tsx

```diff
@@ -0,0 +1,341 @@
+            abortPreviousRequests(
+                () =>
+                    getConfigMapSecretConfigData({
+                        appName,
+                        envName,
+                        resourceId: id,
+                        name: publishedConfigMapSecretData.name,
+                        componentType,
+                        appId: null,
+                        envId: null,
+                        abortControllerRef,
+                    }),
+                abortControllerRef,
+            ),
+        [],
+        draftData.action === DraftAction.Delete &&
+            cmSecretStateLabel === CM_SECRET_STATE.OVERRIDDEN &&
+            componentType === CMSecretComponentType.Secret &&
+            !publishedConfigMapSecretData?.unAuthorized,
+    )
+
+    const protectedSecretData = useMemo<CMSecretConfigData>(() => {
+        if (!protectedSecretDataResLoading && protectedSecretDataRes) {
+            const { data } = protectedSecretDataRes.secretsData
+            if (data.configData?.length) {
+                return { ...publishedConfigMapSecretData, unAuthorized: false }
+            }
+
+            return null
+        }
+
+        return null
+    }, [protectedSecretDataResLoading, protectedSecretDataRes])
+
+    const configMapSecretData = useMemo<CMSecretConfigData>(() => {
+        if (
+            selectedProtectionViewTab === ProtectConfigTabsType.PUBLISHED &&
+            cmSecretStateLabel === CM_SECRET_STATE.UNPUBLISHED
+        ) {
+            return null
+        }
+
+        return draftData.action === DraftAction.Delete &&
+            cmSecretStateLabel === CM_SECRET_STATE.OVERRIDDEN &&
+            componentType === CMSecretComponentType.Secret
+            ? protectedSecretData
+            : { ...JSON.parse(draftData.data).configData?.[0], unAuthorized: !draftData.isAppAdmin }
+    }, [selectedProtectionViewTab, draftData, cmSecretStateLabel, publishedConfigMapSecretData, protectedSecretData])
+
+    const currentConfigMapSecretData = useMemo(
+        () => ({
+            unAuthorized: configMapSecretData?.unAuthorized,
+            ...(formDataRef.current
+                ? (getConfigMapSecretPayload(resolvedFormData ?? formDataRef.current) as ConfigDatum)
+                : configMapSecretData),
+        }),
+        [configMapSecretData],
+    )
+
+    // METHODS
+    const handleCompareFromOptionSelection = (option: SelectPickerOptionType) =>
+        setCompareFromSelectedOptionValue(option.value as CompareFromApprovalOptionsValuesType)
+
+    const getConfigMapSecretDiffViewData = () => {
+        if (draftData.action === DraftAction.Delete) {
+            return isApprovalPendingOptionSelected ? inheritedConfigMapSecretData : null
+        }
+
+        return isApprovalPendingOptionSelected ? configMapSecretData : currentConfigMapSecretData
+    }
+
+    const getCurrentEditorConfig = (): Pick<
+        CompareConfigViewProps,
+        'currentEditorConfig' | 'currentEditorTemplate'
```

Comment:
There is a pending todo to enforce type on its prop type of currentEditorConfig for CM/CS